### ISSUE TITLE
Convert-MarkdownToHTML: Additional Input and Output Capabilities and the Option to Omit Template From Output

### ISSUE THREAD

**Issue by thecliguy** [WetHat/MarkdownToHtml#2]
Firstly, thank you for this superb module. I'm making very good use of it.

Currently `Convert-MarkdownToHTML` only accepts markdown input as a file using the `-Path` parameter and the return value must be output to a file using `-Destination`.

Please consider adding the following:
  * An `-InputObject` parameter.
  * The ability to return an object rather than writing the output to a file, perhaps this could be the default behaviour if `-Destination` is omitted.
  * The ability to omit any template text in the return value, perhaps using a parameter like `-NoTemplate`.

This would enable the command to be used as follows:

```powershell
$String = @'
# Sandbox

| HOSTNAME | OS         | POWERSHELL VER |
|----------|------------|----------------|
| PC123    | Windows 10 | 5.1            |
| PC124    | Windows 7  | 5.0            |
| PC125    | Debian     | 6.1            |

**bold-example**  
_italic-example_

  * A1
    * A2 
  * B1
  * C1
'@

Convert-MarkdownToHTML -InputObject $String -NoTemplate -IncludeExtension "pipetables"

<h1>Sandbox</h1>
<table>
<thead>
<tr>
<th>HOSTNAME</th>
<th>OS</th>
<th>POWERSHELL VER</th>
</tr>
</thead>
<tbody>
<tr>
<td>PC123</td>
<td>Windows 10</td>
<td>5.1</td>
</tr>
<tr>
<td>PC124</td>
<td>Windows 7</td>
<td>5.0</td>
</tr>
<tr>
<td>PC125</td>
<td>Debian</td>
<td>6.1</td>
</tr>
</tbody>
</table>
<p><strong>bold-example</strong><br />
<em>italic-example</em></p>
<ul>
<li>A1
<ul>
<li>A2</li>
</ul>
</li>
<li>B1</li>
<li>C1</li>
</ul>
```

**Comment by WetHat** [comment]
Hi thecliguy,

maybe a new function like:

~~~PowerShell
Convert-MarkdownToHTMLFragment -InputObject $String -IncludeExtension "pipetables"
~~~

would be more elegant? What do you think?

WetHat

**Comment by thecliguy** [comment]
Hello WetHat,

Thumbs up from me, I like the idea of a dedicated function for dealing with fragments, that seems like a clean way to do it.

However, I can't help but feel that `Convert-MarkdownToHTML` would benefit from an `-InputObject` parameter.

Using my own situation as an example, I've a number of existing files from which I extract passages of markdown. If I wanted to use `Convert-MarkdownToHTML` I'd have to write every passage I extract to a file, call `Convert-MarkdownToHTML` with the `-Path` parameter and then delete the file after. Whereas if there were an `-InputObject` parameter, I could pass the extracted passage of markdown to it.

**Comment by WetHat** [comment]
Hi thecliguy,

I think I got your use case. I'm still a bit hesitant to go the `-InputObject`, so here is a new proposal to work with HTML fragments:

### Example 1 - Convert Markdown fragment to html fragment without template expansion
~~~ PowerShell
$markdown | Convert-MarkdownToHTMLFragment -IncludeExtension "pipetables" | Out-File -FilePath $file
# or
Convert-MarkdownToHTMLFragment -InputObject $markdown -IncludeExtension "pipetables" | Out-File -FilePath $file
~~~

### Example 2 - Convert Markdown fragment to html fragment **with** template expansion

~~~ PowerShell
# Using default template
$markdown | Convert-MarkdownToHTMLFragment -IncludeExtension "pipetables" `
| Out-HTMLDocument -title $title -BasePath $base -ChildPath $htmlfile 

# Using custom template
$markdown | Convert-MarkdownToHTMLFragment -IncludeExtension "pipetables" `
| Out-HTMLDocument -title $title -template $templateDir -BasePath $baseDir -ChildPath $htmlfile
~~~

`-BasePath' is the path to  the HTML base directory upon which to build relative URLs to javascript and css resources.
 
What do you think?

WetHat

**Comment by thecliguy** [comment]
Hello WetHat,

Excellent, looks like that should cater for all situations when dealing with fragments.

**Comment by WetHat** [comment]
Should now be addressed by release [1.3.0](https://github.com/WetHat/MarkdownToHtml/releases/tag/v1.3.0)

WetHat
